src/Analyzers/MSTest.Analyzers/Resources.resx

```diff
@@ -181,15 +181,53 @@
   <data name="TestContextShouldBeValidTitle" xml:space="preserve">
     <value>Test context property should have valid layout</value>
   </data>
+  <data name="TestInitializeShouldBeValidDescription" xml:space="preserve">
+    <value>TestInitialize attribute should follow the following layout to be valid:
+- it should be 'public' 
+- it should not be 'static'
+- it should not be generic
+- it should not be 'abstract'
+- it should not take any parameter
+- return type should be 'void' or 'Task'
+- it should not be 'async void'
+- it should not be a special method (finalizer, operator...).</value>
+  </data>
+  <data name="TestInitializeShouldBeValidMessageFormat_NoParameters" xml:space="preserve">
+    <value>Test Initialize '{0}' should not take any parameter</value>
+  </data>
+  <data name="TestInitializeShouldBeValidMessageFormat_NotAbstract" xml:space="preserve">
+    <value>Test initialize '{0}' should not be 'abstract'</value>
+  </data>
+  <data name="TestInitializeShouldBeValidMessageFormat_NotAsyncVoid" xml:space="preserve">
+    <value>Test Initialize '{0}' should not be 'async void'</value>
+  </data>
+  <data name="TestInitializeShouldBeValidMessageFormat_NotGeneric" xml:space="preserve">
+    <value>Test initialize '{0}' should not be generic</value>
+  </data>
+  <data name="TestInitializeShouldBeValidMessageFormat_NotStatic" xml:space="preserve">
+    <value>Test Initialize '{0}' should not be 'static'</value>
+  </data>
+  <data name="TestInitializeShouldBeValidMessageFormat_Ordinary" xml:space="preserve">
+    <value>Test initialize '{0}' should be an 'ordinary' method</value>
+  </data>
+  <data name="TestInitializeShouldBeValidMessageFormat_Public" xml:space="preserve">
+    <value>Test Initialize '{0}' should be 'public'</value>
+  </data>
+  <data name="TestInitializeShouldBeValidMessageFormat_ReturnType" xml:space="preserve">
+    <value>Test Initialize '{0}' should return 'void', 'Task' or 'ValueTask'</value>
+  </data>
+  <data name="TestInitializeShouldBeValidTitle" xml:space="preserve">
+    <value>Test methods should have valid layout</value>
+  </data>
   <data name="TestMethodShouldBeValidDescription" xml:space="preserve">
     <value>Test methods, methods marked with the '[TestMethod]' attribute, should respect the following layout to be considered valid by MSTest:
 - it should be 'public' (or 'internal' if '[assembly: DiscoverInternals]' attribute is set)
 - it should not be 'static'
 - it should not be generic
 - it should not be 'abstract'
-- return type should be 'void' or 'Task'
+- return type should be 'void', 'Task' or 'ValueTask'
 - it should not be 'async void'
-- it should be a special method (finalizer, operator...).</value>
+- it should not be a special method (finalizer, operator...).</value>
```

Comment:
Good catch but I would prefer to have these changes in a different PR please.

Reply:
here we go: https://github.com/microsoft/testfx/pull/2295